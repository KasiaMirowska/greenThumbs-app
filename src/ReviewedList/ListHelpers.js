
import React from 'react';
import ReviewedListItem from '../ReviewedListItem/ReviewedListItem';


const ListHelpers = {
    sortDisplay: (list) => {
        console.log(list)
        const reviews = list.map(pl => {
            return (
                <li className='li' key={pl.id}>
                    <ReviewedListItem
                        placeId={pl.id}
                        yelpId={pl.yelp_id}
                        name={pl.name}
                        city={pl.location_city}
                        category={pl.category}
                        thumbs={pl.checkedThumbs.length}
                    />
                </li>
            )
        })
        return reviews;
    },

}

export default ListHelpers;