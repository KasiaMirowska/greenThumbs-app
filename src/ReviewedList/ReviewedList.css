
.big-container {
    padding-bottom: 5%;
    color: rgb(49, 70, 49);
}
.mid-container {
    padding: 20px;
}

.form2 {
    display: flex; 
}
.section2 {
    display: flex;
}
.section3 {
    display: flex;
    padding-left: 5px;
}
.smaller-header {
    font-family: 'Anton', sans-serif;
    background-color: rgb(207, 236, 236);
    font-size: 20px;
    color: rgb(19, 87, 19);
    margin: 0;
    padding: 20px;
    margin-block-start: 0em;
    margin-block-end: 0em;
    display: flex;
    justify-content: space-between;
}

@media (min-width: 800px) {
    .mid-container {
        padding: 35px;
        width: 70%;
        margin-left: 30%;
    }
    .form2 {
        display: block; 
    }
    
    .section2 {
        padding-left: 15px;
    }
    .section3 {
        display: flex;
        padding-left: 35px;
    }
    .section {
        padding-left: 15px;
        margin: auto;
    }
   
    
    .rev-list {
        /* display:flex; 
        flex-wrap: wrap;
        justify-content: space-evenly; 
        align-items: center;
        justify-content: center;
        flex-flow: row wrap;
        align-content: center; */
    } 

}  
