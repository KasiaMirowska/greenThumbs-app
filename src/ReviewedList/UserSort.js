// import React from 'react';
// import GreenContext from '../Context';
// import ReviewedListItem from '../ReviewedListItem/ReviewedListItem';

// export default class UserSort extends React.Component {
//     static contextType = GreenContext;



//     render () {
        
//         const reviews = this.context.userPlaces.map(pl => {
//             return (
//                 <li className='li' key={pl.id}>
//                     <ReviewedListItem
//                         placeId={pl.id}
//                         yelpId={pl.yelp_id}
//                         name={pl.name}
//                         city={pl.location_city}
//                         category={pl.category}
//                         thumbs={pl.checkedThumbs.length}
//                     />
//                 </li>
//             )
//         })
//         return (
//             <div>
//                 {reviews}
//             </div>
//         )
       
//     }
// }