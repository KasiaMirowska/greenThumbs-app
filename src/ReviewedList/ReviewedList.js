import React from 'react';
import GreenContext from '../Context';
import ReviewedListItem from '../ReviewedListItem/ReviewedListItem';
import './ReviewedList.css';
import GreenCalls from '../Services/GreenCalls';
import ListHelpers from './ListHelpers';

export default class ReviewedList extends React.Component {
    static contextType = GreenContext;
    constructor() {
        super();
        this.state = {
            reviews: [],
            error: null,

        }
    }

    componentDidMount = () => {
        console.log('remounting?')
        GreenCalls.getAllGreenPlacesByUser()
        .then(res => {
            console.log(res, res[0].userid, 'USER PLACES');
            this.context.setCurrentUser(res[0].userid);
            this.context.userSort(res);
        })
        .catch(err => {
            this.setState({
                error: err
            });
        });
        GreenCalls.getAllReviewedPlaces()
            .then(data => {
                this.context.setGreenPlaces(data);
            })
            .then(() => {
                if (this.context.userSelection === true) {
                    this.setState({
                        reviews: ListHelpers.sortDisplay(this.context.userPlaces)
                    })
                } else if (this.context.citySelection === true) {
                    this.setState({
                        reviews: ListHelpers.sortDisplay(this.context.citySortPlaces)
                    })
                } else if (this.context.categorySelection === true) {
                    this.setState({
                        reviews: ListHelpers.sortDisplay(this.context.categorySortPlaces)
                    })
                }
                else {
                    this.setState({
                        reviews: ListHelpers.sortDisplay(this.context.greenPlaces)
                    });
                }

            })
            .catch(err => {
                this.setState({
                    error: err
                });
            })
       
    }

    userSort = (e) => {
        this.props.history.push('/user/places');
        this.context.setUserSelection(true);
    }

    citySort = (e) => {
        e.preventDefault();
        const { city } = e.target;
        this.props.history.push(`/city/${e.target.value}`);
        this.context.citySort(city.value);
        city.value = '';
    }


    handleCategoryInput = (e) => {
        e.preventDefault();
        const category = e.target.value;
        
        this.props.history.push(`/category/${category}`);
        this.context.categorySort(category);
    }

    handleBackButton = (e) => {
        this.props.history.push('/')
        window.location.reload();
    }


    render() {
        const { userSelection, citySelection, categorySelection } = this.context;


        return (
            <div>
                <h2>GREEN thumbsUP reviewed places: </h2>
                
                    {
                        citySelection ? null :
                            <form onSubmit={this.citySort} >
                                <input id='city' type='input' placeholder='enter city' />
                                <button type='submit' >Sort reviews by city</button>
                            </form>
                    }
               
                    {
                        categorySelection ? null :
                        <section>
                            <h3>Sort reviews by category</h3>
                                <select onChange={this.handleCategoryInput} required>
                                    <option value=" ">Choose one </option>
                                    <option value="Coffee-shop">Coffee-shops</option>
                                    <option value="Bakery">Bakeries</option>
                                    <option value="Juice-Bar">Juice-Bars</option>
                                    <option value="Resturant">Restaurants</option>
                                    <option value="Breakfast">Breakfast</option>
                                    <option value="Lunch">Lunch</option>
                                    <option value="Dinner">Dinner</option>
                                </select>
                            </section>
                    }
               
                    {
                        userSelection ? null :
                            <div>
                                <h2>or</h2>
                                <button onClick={this.userSort} >Show only my reviews</button>
                            </div>
                    }
               
                <ul className='rev-list'>
                    {this.state.reviews}
                </ul>
                <button onClick={this.handleBackButton}>back to full list</button>

            </div>
        );
    }
}