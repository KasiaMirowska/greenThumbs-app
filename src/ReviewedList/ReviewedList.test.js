import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter as BR} from 'react-router-dom';
import ReviewedList from './ReviewedList';

describe('List component', () => {
    const props = {
        history: {push: '/'}
    }
   
    it ('renders without crashing',() => {
        const div = document.createElement('div');
        ReactDom.render(<BR><ReviewedList  {...props} /></BR>, div);
        ReactDom.unmountComponentAtNode(div);
    });
   
    it('renders UI as expected', () => {
        const item= renderer.create(<BR><ReviewedList {...props} /></BR>);
        expect(item.toJSON()).toMatchSnapshot();
    })
})