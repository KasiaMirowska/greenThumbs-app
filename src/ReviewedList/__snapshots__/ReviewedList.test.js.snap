// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders UI as expected 1`] = `
<div
  className="big-container list"
>
  <div
    className="smaller-header"
  >
    <h2>
      GREEN
      <span
        className="thumbs"
      >
        thumbs
      </span>
      UP reviewed places: 
    </h2>
  </div>
  <div
    className="error"
  />
  <div
    className="mid-container"
  >
    <div />
    <section
      className="section2"
    >
      <button
        onClick={[Function]}
      >
        Back to full list
      </button>
    </section>
  </div>
  <ul
    className="rev-list"
  />
</div>
`;
